dev/source-release.sh

```diff
@@ -66,7 +66,12 @@ tarball=$tag.tar.gz
 
 # be conservative and use the release hash, even though git produces the same
 # archive (identical hashes) using the scm tag
-git archive $release_hash --prefix $tag/ -o $tarball .baseline api arrow bundled-guava common core data dev flink gradle gradlew hive mr orc parquet pig spark spark2 spark-runtime spark3 spark3-runtime LICENSE NOTICE README.md build.gradle baseline.gradle deploy.gradle tasks.gradle jmh.gradle gradle.properties settings.gradle versions.lock versions.props version.txt
+adds=" .baseline"  # prefixed with a blank space for each file name including the first one
+excludes="build|examples|jitpack.yml|python|site"  # excluded as they are not of use for releasing jars
+echo WARNING! The following files/directories are to be excluded from git archive: ${excludes}
```

Comment:
Can you please use quotes with `echo`? I'm not sure which shell you're using, but I suspect that this isn't going to be portable.

i.e. `echo "WARNING! The following files/directories are to be excluded from git archive: ${excludes}"`. 

Additionally, and this is more of a nit, but I would use `NOTE` over `WARNING`. To me, this is more of an info (or even debug) level log of sorts, and the use of `WARNING` might be confusing for someone simply glancing over it. Again, more of a nit really, but the quotes for `echo` are a best practice (if not typically required in many shells for correctness reasons.).

Reply:
I agree with this, and I would make the message even simpler since everything is basically at the INFO level:

```
echo "Archive exclude list: ${excludes}"
```